homeassistant/components/homekit/type_humidifiers.py

```diff
@@ -0,0 +1,241 @@
+        """
+        if self.linked_humidity_sensor:
+            async_track_state_change(
+                self.hass,
+                self.linked_humidity_sensor,
+                self._async_update_current_humidity,
+            )
+
+        await super().run_handler()
+
+    @callback
+    def _async_update_current_humidity(
+        self, entity_id=None, old_state=None, new_state=None
+    ):
+        """Handle linked humidity sensor state change to update HomeKit value."""
+        if new_state is None:
+            return
+        try:
+            current_humidity = float(new_state.state)
+            if self.char_current_humidity.value != current_humidity:
+                self.char_current_humidity.set_value(current_humidity)
+                _LOGGER.debug(
+                    "%s: Linked humidity sensor %s changed to %d",
+                    self.entity_id,
+                    self.linked_humidity_sensor,
+                    current_humidity,
+                )
+        except ValueError as ex:
+            _LOGGER.error(
+                "%s: Unable to update from linked humidity sensor %s: %s",
+                self.entity_id,
+                self.linked_humidity_sensor,
+                ex,
+            )
+
+    def _set_chars(self, char_values):
+        _LOGGER.debug("HumidifierDehumidifier _set_chars: %s", char_values)
+
+        if CHAR_ACTIVE in char_values:
+            self.call_service(
+                DOMAIN,
+                SERVICE_TURN_ON if char_values[CHAR_ACTIVE] else SERVICE_TURN_OFF,
+                {ATTR_ENTITY_ID: self.entity_id},
+                f"{CHAR_ACTIVE} to {char_values[CHAR_ACTIVE]}",
+            )
+        if self._target_humidity_char_name in char_values:
+            humidity = round(char_values[self._target_humidity_char_name])
+            self.call_service(
+                DOMAIN,
+                SERVICE_SET_HUMIDITY,
+                {ATTR_ENTITY_ID: self.entity_id, ATTR_HUMIDITY: humidity},
+                f"{self._target_humidity_char_name} to {char_values[self._target_humidity_char_name]}{UNIT_PERCENTAGE}",
+            )
+        if CHAR_TARGET_HUMIDIFIER_DEHUMIDIFIER in char_values:
+            # We support either humidifiers or dehumidifiers. If incompatible setting is requested
+            # then we switch off the device to avoid humidifying when the user wants to dehumidify and vice versa
+            hk_value = char_values[CHAR_TARGET_HUMIDIFIER_DEHUMIDIFIER]
+            if (self._hk_device_class == HC_HUMIDIFIER and hk_value == 0) or (
+                self._hk_device_class == HC_DEHUMIDIFIER and hk_value == 1
+            ):
+                self.call_service(
+                    DOMAIN,
+                    SERVICE_TURN_OFF,
+                    {ATTR_ENTITY_ID: self.entity_id},
+                    f"{CHAR_TARGET_HUMIDIFIER_DEHUMIDIFIER} to {char_values[CHAR_TARGET_HUMIDIFIER_DEHUMIDIFIER]}",
+                )
+
+    def get_humidity_range(self):
```

Comment:
```suggestion
    def get_humidity_range_from_state(self, state):
```

It looks like state is fetched a few times during setup the the accessory.  It is faster to fetch it as few times as possible.